marketing/views/marketing_attributable_dau.view.lkml

```diff
@@ -0,0 +1,235 @@
+  }
+
+  dimension: quarter_abr {
+    description: "this dimension will help us trend period over period analysis for QR"
+    type:  string
+    hidden: no
+    view_label: "Period over Period"
+    sql: CASE WHEN FORMAT_DATE("%m",  DATE_TRUNC(${TABLE}.submission_date, QUARTER)) = "01" then "Q1"
+              WHEN FORMAT_DATE("%m",  DATE_TRUNC(${TABLE}.submission_date, QUARTER)) = "04" then "Q2"
+              WHEN FORMAT_DATE("%m",  DATE_TRUNC(${TABLE}.submission_date, QUARTER)) = "07" then "Q3"
+              ELSE "Q4" end;;
+  }
+
+
+  parameter: compare_to {
+    view_label: "Period over Period"
+    description: "Select the templated previous period you would like to compare to. Must be used with Current Date filter"
+    label: "2. Compare To:"
+    type: unquoted
+    # allowed_value: {
+    #   label: "Previous Period"
+    #   value: "Period"
+    # }
+    allowed_value: {
+      label: "Previous Week"
+      value: "Week"
+    }
+    allowed_value: {
+      label: "Previous Month"
+      value: "Month"
+    }
+    allowed_value: {
+      label: "Previous Quarter"
+      value: "Quarter"
+    }
+    allowed_value: {
+      label: "Previous Year"
+      value: "Year"
+    }
+    default_value: "Year"
+  }
+
+  dimension: first_date_in_period {
+    description: "For a well defined period (YoY, QoQ, MoM, WoW), we use date trunc to get the period start date, for arbitrary period modify this dimension to use date sub and number of days in period"
+    type: date
+    hidden: yes
+    sql: DATE_TRUNC(${filter_end_date}, {% parameter compare_to %});;
+  }
+
+  dimension: period_2_start {
+    hidden:  yes
+    description: "Calculates the start of the previous period"
+    type: date
+    sql:
+        DATE_SUB(${first_date_in_period}, INTERVAL 1 {% parameter compare_to %});;
+    convert_tz: no
+  }
+
+  dimension: period_2_end {
+    hidden:  yes
+    description: "Calculates the end of the previous period"
+    type: date
+    sql:
+        DATE_SUB(${filter_end_date}, INTERVAL 1 {% parameter compare_to %});;
+    convert_tz: no
+  }
+
+
+  dimension: period_filtered_measures {
+    hidden: yes
+    description: "We just use this to create the measures for the respective periods (this = current period, last = previous period)"
+    type: string
+    sql:
+        {% if current_date._is_filtered %}
+            CASE
+            WHEN DATE(${TABLE}.submission_date) BETWEEN DATE(${first_date_in_period}) AND DATE(${filter_end_date}) THEN 'this'
+            WHEN DATE(${TABLE}.submission_date) between ${period_2_start} and ${period_2_end} THEN 'last' END
+        {% else %} NULL {% endif %} ;;
+  }
+
+  dimension: app_name {
+    description: "This will help us only the month ending date for DAU"
+    sql: ${TABLE}.app_name ;;
+    type: string
+  }
+
+  dimension: country {
+    description: "This will help us only the month ending date for DAU"
+    sql: ${TABLE}.country ;;
+    type: string
+  }
+
+
+  dimension: paid_vs_organic {
```

Comment:
I wonder, should this specific bit of logic be encoded into the BQ view itself rather than here?